R/utils-latency.R

```diff
@@ -135,52 +135,95 @@ set_asof <- function(new_data, info) {
-  time_values <- new_data %>%
+  # get the minimum value across the checked epi_keys' maximum time values
+  max_time <- new_data %>%
     select(all_of(original_columns)) %>%
     drop_na() %>%
-    pull(time_value)
-  if (length(time_values) <= 0) {
-    cli::cli_abort("the `time_value` column of `new_data` is empty")
-  }
-  as_of <- attributes(new_data)$metadata$as_of
-  max_time <- max(time_values)
+    {
+      # null and "" don't work in `group_by`
+      if (!is.null(epi_keys_checked) && epi_keys_checked != "") {
+        group_by(., get(epi_keys_checked))
+      } else {
+        .
+      }
+    } %>%
+    summarize(time_value = max(time_value)) %>%
+    pull(time_value) %>%
+    min()
+  forecast_date <- attributes(new_data)$metadata$as_of
   # make sure the as_of is sane
-  if (!inherits(as_of, class(time_values)) & !inherits(as_of, "POSIXt")) {
+  if (!inherits(forecast_date, class(max_time)) & !inherits(forecast_date, "POSIXt")) {
     cli::cli_abort(paste(
-      "the data matrix `as_of` value is {as_of}, ",
+      "the data matrix `forecast_date` value is {forecast_date}, ",
       "and not a valid `time_type` with type ",
       "matching `time_value`'s type of ",
-      "{typeof(new_data$time_value)}."
+      "{class(max_time)}."
     ))
   }
-  if (is.null(as_of) || is.na(as_of)) {
+  if (is.null(forecast_date) || is.na(forecast_date)) {
     cli::cli_warn(paste(
-      "epi_data's `as_of` was {as_of}, setting to ",
+      "epi_data's `forecast_date` was {forecast_date}, setting to ",
       "the latest time value, {max_time}."
     ))
-    as_of <- max_time
-  } else if (as_of < max_time) {
+    forecast_date <- max_time
+  } else if (forecast_date < max_time) {
     cli::cli_abort(paste(
-      "`as_of` ({(as_of)}) is before the most ",
+      "`forecast_date` ({(forecast_date)}) is before the most ",
       "recent data ({max_time}). Remove before ",
       "predicting."
     ))
   }
   # TODO cover the rest of the possible types for as_of and max_time...
-  if (class(time_values) == "Date") {
-    as_of <- as.Date(as_of)
+  if (class(max_time) == "Date") {
+    forecast_date <- as.Date(forecast_date)
   }
-  return(as_of)
+  return(forecast_date)
 }
 
 #' the latency is also the amount the shift is off by
 #' @param sign_shift integer. 1 if lag and -1 if ahead. These represent how you
 #'   need to shift the data to bring the 3 day lagged value to today.
 #' @keywords internal
-get_latency <- function(new_data, as_of, column, shift_amount, sign_shift) {
+get_latency <- function(new_data, forecast_date, column, shift_amount, sign_shift, epi_keys_checked) {
   shift_max_date <- new_data %>%
-    drop_na(all_of(column)) %>%
+    drop_na(all_of(column))
+  # null and "" don't work in `group_by`
+  if (!is.null(epi_keys_checked) && epi_keys_checked != "") {
+    shift_max_date <- shift_max_date %>% group_by(get(epi_keys_checked))
+  }
+  shift_max_date <- shift_max_date %>%
+    summarize(time_value = max(time_value)) %>%
```

Comment:
only `summarise()` is in the namespace, not `summarize()`

Reply:
there's a number of other ones, renamed any instance I found for consistency